plugins/source/snyk/client/spec.go

```diff
@@ -21,11 +29,92 @@ type Spec struct {
 
 	// RetryDelaySeconds is an optional parameter to override the default backoff time for retryable requests.
 	RetryDelaySeconds int `json:"retry_delay_seconds,omitempty"`
+
+	TableOptions TableOptions `json:"table_options,omitempty"`
+}
+
+type TableOptions struct {
+	SnykReportingIssues SnykReportingIssuesOptions `json:"snyk_reporting_issues"`
+}
+
+func (t TableOptions) Validate() error {
+	if t.SnykReportingIssues.From != "" {
+		_, err := time.Parse("2006-01-02", t.SnykReportingIssues.From)
+		if err != nil {
+			return fmt.Errorf("invalid from date: %w", err)
+		}
+	}
+	if t.SnykReportingIssues.To != "" {
+		_, err := time.Parse("2006-01-02", t.SnykReportingIssues.To)
+		if err != nil {
+			return fmt.Errorf("invalid to date: %w", err)
+		}
+	}
+	if t.SnykReportingIssues.Period != "" {
+		if !reValidPeriod.MatchString(t.SnykReportingIssues.Period) {
+			return fmt.Errorf("invalid period: %s", t.SnykReportingIssues.Period)
+		}
+	}
+	if t.SnykReportingIssues.To != "" && t.SnykReportingIssues.From == "" {
```

Comment:
Shouldn't it be flipped?
Judging by `SnykReportingIssuesOptions.FromTime`, `From == "" && Period != ""` is fine

Reply:
I think it's correct; this case prevents setting `to` if `from` is not also set. We accept these combinations:
 - `from` + `to`
 - `from` (default `to` is now)
 - `period` (relative to now)
 
We do not accept `period` combined with either `to` or `from`, and we don't accept `to` without `from`.  Does that make sense? Do you still think there is something wrong here?